### PR TITLE
Danielle - Adagrams JS

### PR DESCRIPTION
**Pull request by danimetz**
# JS Adagrams
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
--- | ---
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? |  The loops, but instead of having the handy ruby methods available I created the functionality using Javascript loops 
Did you need to use different strategies to find information online about JavaScript than you would for Ruby? | Not really just needed to look up specific things for JS and I used the Javascript MDN
What was a challenge you were able to overcome on this assignment? |  All the curly brackets! It was good to work through this assignment to get a better handle on the JS syntax.  I feel a little bit more comfortable with loops and I'm sure I'll continue to get better throughout this unit.
What is your favorite thing about learning a new programming language? | That I can read and understand what is happening
What is your least favorite thing about learning a new programming language? |  Getting the right syntax, and debugging without pry
Do you have any recommendations on how we could improve this project for the next cohort? | If there was an easier way to debug our loops without having a million console logs that would be super helpful. I also would have liked to know that I couldn't just use a return to break out of the loop I needed to assign my value to a variable to return at the very end otherwise I wouldn't get what I was expecting out of my loop.

### CODE REVIEW

File: src/adagrams.js [hunk 1]
```diff
@@ -1,8 +1,142 @@
 const Adagrams = {
   drawLetters() {
     // Implement this method for wave 1
+    let letters = {
```
Comment (CheezItMan) [inline src/adagrams.js:4]: Prefer `const` over `let`

File: src/adagrams.js [hunk 2]
```diff
@@ -1,8 +1,142 @@
...
+      "U" : 4,
+      "V" : 2,
+      "W" : 2,
+      "X" : 1,
+      "Y" : 2,
+      "Z" : 1
+    };
+
+    //Creates the array of all the letters available to pick
+    let letterProbability = [];
```
Comment (CheezItMan) [inline src/adagrams.js:34]: Again, prefer using `const` instead of `let`.

**Comment by CheezItMan** [comment]
# JS Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions |  Check, you'll have an easier time debugging in the browser.  
Small commits with meaningful commit messages |  Only **one** commit!  Gasp!  
**Code Requirements** |
`drawLetters` method |  Check
Uses appropriate data structure to store the letter distribution |Check
All tests for `drawLetters` pass |Check
`usesAvailableLetters` method |Check
All tests for `usesAvailableLetters` pass |Check
`scoreWord` method |Check
Uses appropriate data structure to store the letter scores |Check
All tests for `scoreWord` pass |Check
`highestScoreFrom` method |Check
Appropriately handles edge cases for tie-breaking logic |Check
All tests for `highestScoreFrom` pass |Check
**Overall** |  Well done, you hit all the learning goals for the project.  I do suggest you use `const` instead of `let` wherever you can.  Otherwise nicely done.
